### PR TITLE
workflows: Replace deprecated ::set-env

### PR DESCRIPTION
**Pull request by jepler**
Closes: #3526

### CODE REVIEW

**Comment by jepler** [comment]
This isn't right, because the place where I can spot the environment variable being used, it gives blank instead of the right value:
```
Run sphinx-build -E -W -b html -D version= -D release= . _build/html
```

**Comment by jepler** [comment]
Next iteration was right: `Run sphinx-build -E -W -b html -D version=6.0.0-beta.2-33-g5ae2423c4 -D release=6.0.0-beta.2-33-g5ae2423c4 . _build/html`

However, the important "upload to s3" steps can't be 100% checked until after this is merged.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -132,7 +132,7 @@ jobs:
     - name: Install dependencies
       run: |
         brew install gettext awscli
-        echo "::set-env name=PATH::/usr/local/opt/gettext/bin:$PATH"
+        echo >>$GITHUB_PATH /usr/local/opt/gettext/bin
```
Comment (hathach) [inline .github/workflows/build.yml:135]: not important at all, though the github doc https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
use the syntax that put ` >> $GITHUB_ENV` at the end 

```
echo "action_state=yellow" >> $GITHUB_ENV
echo "/path/to/dir" >> $GITHUB_PATH
```

Out of curiosity, why would you change the order, or maybe they update the doc recently ?

Comment (hathach) [reply .github/workflows/build.yml:135]: right, it makes lots of sense :+1:

**Comment by tannewt** [comment]
Build failure is a clone issue and unrelated.

**Comment by jepler** [comment]
artifacts are uploading correctly after this was merged, and the warnings are gone from the overview page.

**Pull request merged**